consumer.go

```diff
@@ -195,15 +199,36 @@ func (c *segmentConsumer) positionOf(m *kafka.Message) ([]byte, error) {
 	return p.json()
 }
 
-func (c *segmentConsumer) Ack() error {
-	// See issue related to this: https://github.com/ConduitIO/conduit-connector-kafka/issues/23
-	err := c.reader.CommitMessages(context.Background(), *c.lastMsgRead)
+func (c *segmentConsumer) Ack(position sdk.Position) error {
+	err := c.canAck(position)
+	if err != nil {
+		return fmt.Errorf("ack not possible: %w", err)
+	}
+	err = c.reader.CommitMessages(context.Background(), *c.unackMessages[0])
 	if err != nil {
 		return fmt.Errorf("couldn't commit messages: %w", err)
 	}
+	// remove the message from slice of unacknowledged messages
+	c.unackMessages = c.unackMessages[1:]
 	return nil
 }
 
+func (c *segmentConsumer) canAck(position sdk.Position) error {
+	if len(c.unackMessages) == 0 {
+		return fmt.Errorf("requested ack for %q but no unacknowledged messages found", position)
+	}
+	pos, err := c.positionOf(c.unackMessages[0])
+	if err != nil {
+		return fmt.Errorf("failed to get position of first unacknowledged message: %w", err)
+	}
+	// We're going to yell at Conduit for not keeping its promise:
+	// acks should be requested in the same order reads were done.
+	if bytes.Compare(pos, position) != 0 {
```

Comment:
Depending on how we solve nacks, this might not always be the expected behavior. Let's talk about this synchronously, since it's a bit complicated to fit into a PR comment, also this PR is not the correct place to have the debate.

Reply:
We decided to implement a guarantee in Conduit that will ensure acknowledgments are sent in the correct order and won't be skipped. This code should work fine once we add that.